metrics/cli/docs/features/update/06_update_feature.md

```diff
@@ -165,4 +175,139 @@ Having considered all the complex points and edge cases, we conclude that the `U
+
+Let's review the component diagram that shows how the main component interacts with each other:
+
+![Update command component diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/Flank/flank-dashboard/raw/update_feature_design/metrics/cli/docs/features/update/diagrams/update_command_components_diagram.puml)
+
+## User Interface
+>How users will interact with the feature (API, CLI, Graphical interface, etc.).
+
+The `Update feature` implies adding a new command to the `Metrics CLI` application. Users can interact with the update feature as with any other command of the Metrics CLI. This interaction is based on the following rule: `metrics <command> [arguments]`, where `metrics` is the Metrics CLI executable name, `<command>` is a command to execute name, and `[arguments]` are arguments to pass to the command.
+
+The following snippet demonstrates the Metrics command-line user interface for the `--help` option of the `Metrics CLI` tool after the `Update feature` integration into this CLI.
+
+```diff
+Metrics CLI.
+
+Usage: metrics <command> [arguments]
+
+Global options:
+  -h, --help    Print this usage information.
+  
+Available commands:
+  deploy   Creates GCloud and Firebase projects for Metrics components and deploys the Metrics Web Application.
+  doctor   Shows the version information of the third-party dependencies.
++ update   Updates the Metrics components and redeploys the Metrics Web Application to the Firebase project.
+
+Run "metrics help <command>" for more information about a command.
+```
+
+In the example above, we can notice that a new `update` command has been added with its description accordingly. 
+For more information on how the `update` command works, the user can call it with the `--help` option. The following snippet demonstrates the command-line user interface for this:
+
+```bash
+Usage: metrics update [arguments]
+-h, --help                         Print this usage information.
+    --config-file=<config.yaml>    A path to the YAML configuration file.
+
+Run "metrics help" to see global options.
+```
+
+As we can notice, the interface for the update feature is pretty similar to other features of the Metrics CLI. Thus, since we clarified the usage, let's proceed to a deeper review of the feature structure and implementation details.
+
+## Program
+> Detailed solution description to class/method level.
+
+The update feature implies two main actions we'd like to review in the below subsections:
+- [Parsing YAML configuration file](#parsing-yaml-configuration-file)
+- [Redeploy Metrics components](#redeploy-metrics-components)
+
+### Parsing YAML configuration file
+
+The YAML configuration file for the update feature has the following structure:
+
+![Update config diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/Flank/flank-dashboard/raw/update_feature_design/metrics/cli/docs/features/update/diagrams/update_config_yaml_diagram.puml)
```

Comment:
Should we use here a simple snippet from configuration_template (without detailed documentation) as this diagram could look a bit confusing due to 2 rows layout where a reader could think that the second column somehow explains what value from the first column means? :readable: